include/openenclave/tracee.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_TRACE_E_H
+#define _OE_TRACE_E_H
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/enclave.h>
+#include <openenclave/log.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * Define callback to process logs generated by enclave.
+ *
+ * The callback, defined as expected, will be called every time OE_TRACE is
+ * is called by enclave routines, irrespective of log level. The callback will
+ * be processed under log lock.
+ *
+ * NOTE: The function oe_enclave_log_set_callback sets the callback and only
+ * then enclave logs will be processed through the callback function.
+ *
+ * @param[in] context The parameter can be used by the user to send in
+ * additional variables/ structures that might be needed to process the logs.
+ * @param[in] level The parameter holds the log level of the current enclave
+ * log being processed and can be used by the callback.
+ * @param[in] thread_id The parameter holds the thread id of the curent
+ * processing thread.
+ * @param[in] message The parameter holds the enclave log message generated.
+ */
+typedef void (*oe_enclave_log_callback_t)(
+    void* context,
+    oe_log_level_t level,
+    uint64_t thread_id,
+    const char* message);
+
+/**
+ * Set callback that will process logs generated by enclave.
+ *
+ * This function, when called ensures the enclave logs are transfered to
+ * the preferred callback instead of the regular route.
+ *
+ * @param[in] context The parameter holds a pointer to variable/
+ * structure that can be used to process the logs inside the callback
+ * function.
+ * @param[in] callback The parameter holds pointer to function that is
+ * be assigned as the callback to process enclave logs.
+ *
+ * @returns OE_OK successful
+ * @returns OE_FAILURE failed to add callback
```

Comment:
```suggestion
 * @retval OE_OK The operation was successful.
 * @retval OE_FAILURE Failed to add callback.
```
Use retval for discrete values.